File: Java/Recursive descent parser/7/src/expression/And.java
package expression;

import expression.exceptions.BinaryOperation;

public class And extends BinaryOperation {

    public And(TripleExpression left, TripleExpression right) {
        super(left, right);
    }

    public int eval(int l, int r) {
        return l & r;
    }

    public double eval(double l, double r) {
        return (int) l & (int) r;
    }
}

File: Mathlog/B-D/src/expression/And.kt
package expression

class And(left : Expression, right: Expression) : BinaryOperator(left, right, "&") {

    override fun getNewExpr(letterToExpr: Map<String, Expression>): Expression {
        return And(left.getNewExpr(letterToExpr),right.getNewExpr(letterToExpr))
    }
}